modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v2_0/Swagger2Translator.java

```diff
@@ -581,4 +617,523 @@ public static Swagger getSwagger(Definition definition) {
+        fillRwadefResponses(swaggerOperation, operation, contract);
+        fillInputPayload(swaggerOperation, operation, contract);
+
+        resource.getOperations().add(operation);
+    }
+
+    private static void fillRwadefParameters(Operation swaggerOperation,
+            org.restlet.ext.apispark.internal.model.Operation operation, Resource resource,
+            Map<String, org.restlet.ext.apispark.internal.model.Parameter> parameters) {
+        if (swaggerOperation.getParameters() == null) {
+            return;
+        }
+
+        for (Parameter parameter : swaggerOperation.getParameters()) {
+            if (parameter instanceof QueryParameter) {
+                org.restlet.ext.apispark.internal.model.QueryParameter queryParameter =
+                        new org.restlet.ext.apispark.internal.model.QueryParameter();
+                QueryParameter swaggerQueryParameter = (QueryParameter) parameter;
+
+                fillRwadefQueryParameter(queryParameter, swaggerQueryParameter);
+                operation.getQueryParameters().add(queryParameter);
+            } else if (parameter instanceof PathParameter) {
+                org.restlet.ext.apispark.internal.model.PathVariable pathVariable =
+                        new org.restlet.ext.apispark.internal.model.PathVariable();
+                PathParameter swaggerPathVariable = (PathParameter) parameter;
+
+                fillRwadefPathVariable(pathVariable, swaggerPathVariable);
+                if (resource.getPathVariable(pathVariable.getName()) == null) {
+                    resource.getPathVariables().add(pathVariable);
+                }
+            }
+        }
+    }
+
+    private static void fillRwadefResponses(Operation swaggerOperation,
+            org.restlet.ext.apispark.internal.model.Operation operation,
+            Contract contract) {
+        if (swaggerOperation == null) {
+            return;
+        }
+
+        for (String key : swaggerOperation.getResponses().keySet()) {
+            Response swaggerResponse = swaggerOperation.getResponses().get(key);
+            org.restlet.ext.apispark.internal.model.Response response =
+                    new org.restlet.ext.apispark.internal.model.Response();
+
+            Integer statusCode = null;
+            try {
+                statusCode = Integer.parseInt(key);
+                response.setCode(statusCode);
+            } catch (Exception e) {
+                // TODO: what to do with "Default" responses ?
+                LOGGER.warning("Response " + key + " for operation " + swaggerOperation.getOperationId() +
+                        " could not be retrieved because its key is not a valid status code.");
+                continue;
+            }
+
+            response.setMessage(swaggerResponse.getDescription());
+            response.setName(statusCode.toString().matches("2[0-9]{2}") ? "Success" : "Error " + statusCode);
```

Comment:
same comments on other kind of status. Could factorize this code with code above?